x/auth/tx/service.go

```diff
@@ -62,57 +57,36 @@ func (s txServer) GetTxsEvent(ctx context.Context, req *txtypes.GetTxsEventReque
 		return nil, status.Error(codes.InvalidArgument, "must declare at least one event to search")
 	}
 
-	tmEvents := make([]string, len(req.Events))
-	for i, event := range req.Events {
-		if !strings.Contains(event, "=") {
-			return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("invalid event; event %s should be of the format: %s", event, eventFormat))
-		} else if strings.Count(event, "=") > 1 {
+	for _, event := range req.Events {
+		if !strings.Contains(event, "=") || strings.Count(event, "=") > 1 {
 			return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("invalid event; event %s should be of the format: %s", event, eventFormat))
 		}
-
-		tokens := strings.Split(event, "=")
-		if tokens[0] == tmtypes.TxHeightKey {
-			event = fmt.Sprintf("%s=%s", tokens[0], tokens[1])
-		} else {
-			event = fmt.Sprintf("%s='%s'", tokens[0], tokens[1])
-		}
-
-		tmEvents[i] = event
 	}
 
-	query := strings.Join(tmEvents, " AND ")
-
-	result, err := s.clientCtx.Client.TxSearch(ctx, query, false, &page, &limit, "")
+	result, err := queryTxsByEvents(s.clientCtx, req.Events, page, limit, "")
```

Comment:
Calling `authclient.QueryTxsByEvents` introduces an import cycle, so I just copy/pasted `queryTxsByEvents` to this package :(